feat(max): expose max context from scene logics

kappa90 opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem
When navigating to a dashboard, we want the dashboard to automatically show in Max's UI context.

## Changes
- NOTE: The PR has been completely re-done based on @Twixes 's feedback. Now logics expose a `maxContext` object which is picked up by `maxContextLogic` automatically
- Scenes that support context: homepage, dashboards, insights
- Supported entities: dashboards, insights (only supported query types), events, actions
- Added READMEs to explain how to support more query and entity types for context
- Removed 'Current Page' from context options
- Bonus: hide Max on non-product pages (onboarding, signup, etc.)
 
## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [X] No docs needed for this change

## How did you test this code?
Added/Fixed tests

Commit:
feat: auto add active dashboard to context

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

Commit:
fix: better typing + tests + fixed logic

Commit:
Merge remote-tracking branch 'origin/feat/auto-add-dashboard' into feat/auto-add-dashboard

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

Commit:
fix: types

Commit:
Merge remote-tracking branch 'origin/feat/auto-add-dashboard' into feat/auto-add-dashboard

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

Commit:
Merge remote-tracking branch 'origin/master' into feat/auto-add-dashboard

Commit:
Merge remote-tracking branch 'origin/feat/auto-add-dashboard' into feat/auto-add-dashboard

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

Twixes reviewed (approved):
Works 🔧 

Although a couple notes around the experience:
- It's surprising that the specific dashboard/insight is included, rather than "Current page", when "Current page" exists as an option. Should we remove "Current page", or make that the auto-included piece of context?
- We're missing a skeleton state for the context while we're waiting for the concrete insight/dashboard to load - the tag pops in suddenly, even though we know from the get-go that it's _about_ to appear

Twixes commented:
Third point:

`dashboardLogic` is now very aware `maxContextLogic` API, which won't be easy to maintain. The dependency should ideally be inverted here – `maxContextLogic` should be the one grabbing at scene logics to collect the relevant context.

I propose we re-architect this similarly to how `breadcrumbsLogic` works with standardized scene logic `breadcrumbs` selectors: Every logic registered in an `export const scene: SceneExport` MAY expose a `maxContext` selector (`MaxContextItem[]`), and doesn't need to know anything else. `maxContextLogic` can then automatically pick that selector up.

Twixes reviewed (changes requested):
Actually before we go further into this direction – let's rearchitect `breadcrumbs`-style, i.e. my third point

Commit:
refactor: expose maxContext from scene logics

Commit:
wip: refactor

Commit:
fix: readme + tests

Commit:
Merge remote-tracking branch 'origin/master' into feat/auto-add-dashboard

Commit:
fix: readme typo

Commit:
fix: readme

Commit:
fix: enums

Commit:
fix: enums

Commit:
fix: remove test / story

Commit:
Merge remote-tracking branch 'origin/master' into feat/auto-add-dashboard

Commit:
fix: update schema

Commit:
fix: bugs

Twixes committed:
Merge branch 'master' into feat/auto-add-dashboard

Twixes committed:
Fix memory leak

We only want to use an already-mounted `dashboardLogic`, because if we mount it _inside_ a selector, it will subsequently _never_ get unmounted.

Twixes reviewed (changes requested):
Found a bug: When I start a new thread with the current page's context auto-added, and then exit that thread – I no longer get the page context auto-added ever again

Twixes commented on frontend/src/scenes/insights/insightSceneLogic.tsx:
```diff
@@ -233,6 +236,18 @@ export const insightSceneLogic = kea<insightSceneLogicType>([
+        maxContext: [
+            (s) => [s.insight, s.insightLogicRef],
+            (insight: Partial<QueryBasedInsightModel>, insightLogicRef: any): MaxContextSelector => {
+                // Only provide insight context when in view mode
+                // In edit mode, MaxAI has the `create_and_query_insight` tool available
+                const isInViewMode = insightLogicRef?.logic.values.isInViewMode
+                if (!insight || !insight.short_id || !insight.query || !isInViewMode) {
+                    return []
+                }
+                return [createMaxContextHelpers.insight(insight)]
```
This genuinely confused me when I was running the app and trying out context in insights – I was like "where has my insight gone". I know it's just ported behavior, but thinking it'd be clearer to not consider edit/view mode here

(Separately: Feeling like the "Edit insight" tool should be available in view mode too – it's un-magical that I have to manually click "Edit" myself to get going)

Commit:
Merge remote-tracking branch 'origin/master' into feat/auto-add-dashboard

Commit:
fix: review

kappa90 commented:
@Twixes Updated type naming to be more consistent and understandable for other devs, fixed the bug you mentioned that would not re-add context when creating a new conversation, and made it so that we still auto-add context when editing insights.

Commit:
fix: rename variables

Twixes committed:
Fix insight.derived_name not being used

Insights with no custom user-defined name were showing up as "Insight abcdef" despite appearing as "Pageview user conversion rate".

Twixes reviewed (approved):
Excellent

Commit:
Merge remote-tracking branch 'origin/master' into feat/auto-add-dashboard

Commit:
Merge remote-tracking branch 'origin/feat/auto-add-dashboard' into feat/auto-add-dashboard

Commit:
Merge branch 'master' into feat/auto-add-dashboard

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (4)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

Twixes committed:
Merge branch 'master' into feat/auto-add-dashboard

Twixes committed:
Rebuild schema files

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

Twixes committed:
Merge branch 'master' into feat/auto-add-dashboard

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
Merge branch 'master' into feat/auto-add-dashboard

github-actions[bot] committed:
Update UI snapshots for `chromium` (12)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)
